libdevcore/CommonData.h

```diff
@@ -42,22 +42,44 @@ enum class WhenError
 	Throw = 1,
 };
 
-enum class HexPrefix
+namespace
 {
-	DontAdd = 0,
-	Add = 1,
-};
-/// Convert a series of bytes to the corresponding string of hex duplets.
-/// @param _w specifies the width of the first of the elements. Defaults to two - enough to represent a byte.
+
+const char* hexdigits = "0123456789abcdef";
+
+template <class T>
+inline void writeHex(T const& _data, std::string& o_hex, size_t _off)
+{
+	for (byte i: _data)
+	{
+		o_hex[_off++] = hexdigits[i>>4];
+		o_hex[_off++] = hexdigits[i&0x0f];
+	}
+}
+
+}
+
+/// Convert a series of bytes to the corresponding hex string.
 /// @example toHex("A\x69") == "4169"
 template <class T>
-std::string toHex(T const& _data, int _w = 2, HexPrefix _prefix = HexPrefix::DontAdd)
+std::string toHex(T const& _data)
+{
+	static_assert(sizeof(typename T::value_type) == 1, "toHex needs byte-sized element type");
+	std::string hex(_data.size()*2, '0');
```

Comment:
Could you move this string declaration into `writeHex` and return it from there? Instead of having it as a input&output parameter which is really confusing